pyro/infer/predictive.py

```diff
@@ -99,7 +100,7 @@ def _predictive(model, posterior_samples, num_samples, return_sites=None,
     return predictions
 
 
-class Predictive:
+class Predictive(PyroModule):
```

Comment:
Hmm, this will currently break in a few ways:
1. in `PyroModule._pyro_set_supermodule()`, `assert not value._pyro_context.used` will fail
2. names will change in `model` and `guide`, invalidating the `PyroModule`-paramstore coupling.

Maybe we should make `Predictive` a `torch.nn.Module` instead?

Reply:
Yes, I bumped into exactly those two issues while debugging. 😞 Let me try your suggestion instead.

Reply:
On further thought, I don't see how `PyroModule` is appropriate. `nn.Module` seems like the right way to serialize. But I don't fully understand the issues...

Reply:
For example if `model` and `guide` have `PyroSample` sites, those addresses must match exactly, and that can't happen in a single `PyroModule`, but is fine if they live in a common `nn.Module`.

Reply:
You are right - the poutine calls should also be serialized correctly. I am chasing some unrelated issues though, still debugging.